jobrunner/executors/local.py

```diff
@@ -553,6 +567,12 @@ def write_job_logs(
             )
 
 
+def write_job_metadata(job_definition, job_metadata):
```

Comment:
Worth maybe yoinking this up to being near to read_job_metadata, as they are inverses of each other? Keep the path logic shared between them closer together.